tests/test_model.py

```diff
@@ -78,14 +78,22 @@ def test_cpes():
 
 
 def test_nevra():
-    no_release = ComponentFactory(release="", version="1")
-    assert not no_release.nvr.endswith("-")
-    assert not PackageURL.from_string(no_release.purl).version.endswith("-")
-    # epoch is a property of Component which retrieves the value for meta_attr
-    no_epoch = ComponentFactory()
-    assert ":" not in no_epoch.nevra
-    no_arch = ComponentFactory(arch="")
-    assert not no_arch.nevra.endswith("-")
+    for component_type in Component.Type.values:
+        no_release = ComponentFactory(
+            name=component_type, release="", version="1", type=component_type
+        )
+        assert not no_release.nvr.endswith("-")
+
+        if component_type == "OCI":
```

Comment:
nit: I have a different fix for this in https://github.com/RedHatProductSecurity/component-registry/pull/215

Reply:
Ack, and thank you! I'll fix the assert to match your version during a rebase, but I think I'll also keep the for loop I added. Right now we only test one type of component randomly, but we have different code for different purl types. So if one of those codepaths is broken, the test will sometimes pass and sometimes fail randomly.

Switching to a for loop improves coverage by testing every code path for every type, and means that tests will pass or fail consistently if something is broken, instead of randomly like they do today.